### PR TITLE
Add smoke test for YaST2 firstboot module

### PR DESCRIPTION
**Pull request by mloviska**
Initial PR to enable YaST2 Firstboot testing.

- Related ticket: [[functional][y] new yast2 firstboot wizard test](https://progress.opensuse.org/issues/40184)
- Needles: 
   * <del>[o3](https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/536)</del>
   * [osd](https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/1105)
   * [leap15.1 - current state](https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/538)
- Verification runs: 
  * firstboot VRs:
     * AY:
        * [autoyast_y2_firstboot@64bit-tw](http://eris.suse.cz/tests/12901)
        * [autoyast_y2_firstboot@64bit-12-SP5](http://eris.suse.cz/tests/12890)
        * [autoyast_y2_firstboot@64bit-15-SP1](http://eris.suse.cz/tests/12900)
    *  Classic:
        * [sle-15-SP1-yast2_firstboot@64bit](http://eris.suse.cz/tests/12891)
        * [sle-12-SP5-yast2_firstboot@64bit](http://eris.suse.cz/tests/12899)
        * [Tumbleweed-yast2_firstboot@64bit](http://eris.suse.cz/tests/12907)
  * regression:
    * [user_settings_root](http://eris.suse.cz/tests/12788#step/user_settings_root/1)
    * [user_settings](http://eris.suse.cz/tests/12789#step/user_settings/1)
  * bugs:
    * [boo#1105792 leap15.1 - branding and missing package in DVD](https://bugzilla.opensuse.org/show_bug.cgi?id=1105792)
    * [License Agreement view is shown in Leap15.1](https://bugzilla.opensuse.org/show_bug.cgi?id=1131301)
    * [ay leap15.1 - missing firstboot rpm in DVD](http://eris.suse.cz/tests/12941#step/installation/29)
    * [branding and license agreement view not skipped](http://eris.suse.cz/tests/12939#step/yast2_firstboot/25)
    * [Bug 1131327 - [Build 0134] openQA test fails in yast2_firstboot - license terms checkbox gets cleared after pressing next button](https://bugzilla.suse.com/show_bug.cgi?id=1131327)

### CODE REVIEW

**Comment by mloviska** [comment]
@rwx788 sure!

File: tests/installation/yast2_firstboot.pm [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    $self->verify_license_has_to_be_accepted;
+    $self->accept_license;
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+    # WA
+    send_key 'alt-a';
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+    #
+}
+
+sub welcome {
```
Comment (rwx788) [inline tests/installation/yast2_firstboot.pm]: Initially I thought that we could reuse existing modules, at least we could do that for some. Have you also considered splitting these into separate test modules?

Comment (mloviska) [reply tests/installation/yast2_firstboot.pm]: yes, and there is not much to be reused from my point of view. For instance, *welcome.pm*. In the classical installer the module handles language and keyboard layout as well as product selection, whereas Firstboot module separates L&K view from Welcome view.

Comment (rwx788) [reply tests/installation/yast2_firstboot.pm]: My main concern is about this module being maintained separately and becoming monster. `welcome.pm` is mixing multiple purposes, but language and keyboard widgets are same I guess. Anyway, we can leave for now.

File: tests/installation/yast2_firstboot.pm [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+sub license {
+    my $self = shift;
+    return unless is_sle('>=12-sp4');
+    assert_screen('license-agreement');
+    $self->verify_license_has_to_be_accepted;
+    $self->accept_license;
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+    # WA
```
Comment (rwx788) [inline tests/installation/yast2_firstboot.pm]: I think that's the bug you've mentioned yesterday. Let's add soft-failure here if it is the case.

File: tests/installation/yast2_firstboot.pm [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    send_key 'alt-a';
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+    #
```
Comment (rwx788) [inline tests/installation/yast2_firstboot.pm]: Was a comment expected here? =)

File: tests/installation/yast2_firstboot.pm (hunk 2 above)
Comment (mloviska) [reply tests/installation/yast2_firstboot.pm]: ah, haven't pushed the changes. Thanks!

**Comment by rwx788** [comment]
Alright, let's move forward. Could you please also update our [test plan](https://gitlab.suse.de/riafarov/qa-sle-functional-y/blob/master/SLES_Integration_Level_Testplan.md), once scenarios are added to the job group?

**Comment by mloviska** [comment]
@rwx788, it is there already. At the end of https://gitlab.suse.de/riafarov/qa-sle-functional-y/blob/master/SLES_Integration_Level_Testplan.md#test-suites and https://gitlab.suse.de/riafarov/qa-sle-functional-y/blob/master/SLES_Integration_Level_Testplan.md#installer

**Comment by rwx788** [comment]
> @rwx788, it is there already. At the end of https://gitlab.suse.de/riafarov/qa-sle-functional-y/blob/master/SLES_Integration_Level_Testplan.md#test-suites and https://gitlab.suse.de/riafarov/qa-sle-functional-y/blob/master/SLES_Integration_Level_Testplan.md#installer

Ah, you are right. Somehow I thought it was only about ibft. Thanks a lot!

**Pull request merged**